M2/Macaulay2/tests/normal/quotientRemainder.m2

```diff
@@ -7,6 +7,14 @@ NA=gens intersect(image(N),image(A));
 (mat1,mat2) = quotientRemainder(NA,N)
 assert(mat2 == 0)
 assert(N * mat1 == NA)
+
+-- test for left and right quotients
+M = coker(5 * id_(ZZ^2));
+f = map(M, M, {{1,2},{3,4}});
+g = map(M, M, {{2,3},{4,0}});
+assert same {g, quotient(f * g, f),  (f * g) // f}
+assert same {f, quotient'(f * g, g), g \\ (f * g)}
+
 end
```

Comment:
Here is another test for left and right quotients.  Can you add this in (or something like it?)
```
S = ZZ/101[a..d]/(a^2+b^2+c^2+d^2)
F = res(coker vars S, LengthLimit => 10)
A = coker F.dd_4
B = coker F.dd_3
C = coker F.dd_5
HAB = Hom(A,B)
basis(0, HAB)
f = homomorphism(basis(0, HAB) * random(S^48, S^1))
HBC = Hom(B,C)
basis(3, HBC)
gd = homomorphism(basis(3, HBC) * random(S^32, S^1))
g = map(target gd ** S^{3}, source gd, gd)
degree g == {0} and isHomogeneous g
h = g*f
-- So: f : A --> B
--     g : B --> C ** S^{3}
--     h : A --> C ** S^{3}
assert(g * (h//g) == h)
assert(h // g == f) 
assert(f \\ h == g)
assert((f\\h) * f == h)
g // h == 0. -- does it always happen that this is zero if lift can't occur?  Probably not
h \\ f == 0
```

Reply:
Btw, I'm about to add a commit that lets you just do `homomorphism random HAB`.

Reply:
Thanks for the test, added it now.